### PR TITLE
Add timestamp for start of turn

### PR DESCRIPTION
**Pull request by Evert-R**
This PR sets a timestamp `start_of_turn` in the result of the first card.
First when the board has been loaded, then on every third click of a turn. 

Closes: #1298

### CODE REVIEW

**Review by BeritJanssen — COMMENTED** [review]
In terms of analyzing data: I think it would be better to not set the `start_of_turn` variable on each card, but instead, send a `start_of_turn` variable within the `result` object. This will be saved to the results' json data. The `response_interval_ms` can probably be removed (check with @Lia-JX-Li ), as it can be derived from the timestamps.

File: frontend/src/components/MatchingPairs/MatchingPairs.tsx [hunk 1]
```diff
@@ -62,6 +62,8 @@ const MatchingPairs = ({
         yPosition.current = posY;
     }
 
+    let startOfTurn = performance.now();
```
Comment (drikusroor) [inline frontend/src/components/MatchingPairs/MatchingPairs.tsx:65]: Does it work like this in React? 🤔 If written like this, I expect `startOfTurn` to be overwritten in every re-render... I would expect a state variable that gets set on initialization and to be set again using `setStartOfTurn` in the `finishTurn` function (same place as you now do the reassignment).

File: frontend/src/components/MatchingPairs/MatchingPairs.tsx [hunk 2]
```diff
@@ -141,7 +142,8 @@ const MatchingPairs = ({
                 currentCard.turned = true;
                 currentCard.noevents = true;
                 currentCard.boardposition = index + 1;
-                currentCard.timestamp = performance.now();
+                setStartOfTurn(performance.now());
+                currentCard.start_of_turn = startOfTurn;
```
Comment (BeritJanssen) [inline frontend/src/components/MatchingPairs/MatchingPairs.tsx:146]:
```suggestion
```

File: frontend/src/components/MatchingPairs/MatchingPairs.tsx [hunk 3]
```diff
@@ -152,6 +154,7 @@ const MatchingPairs = ({
     };
 
     const finishTurn = () => {
+        setStartOfTurn(performance.now());
```
Comment (BeritJanssen) [inline frontend/src/components/MatchingPairs/MatchingPairs.tsx:157]: Why set this here and at line 145? Wouldn't it be enough to do it in one of those two places?

File: frontend/src/components/MatchingPairs/MatchingPairs.tsx (hunk 1 above)
Comment (Evert-R) [reply frontend/src/components/MatchingPairs/MatchingPairs.tsx]: It did work but using a state variable would indeed be better.

**Comment by Evert-R** [comment]
> In terms of analyzing data: I think it would be better to not set the `start_of_turn` variable on each card, but instead, send a `start_of_turn` variable within the `result` object. This will be saved to the results' json data. The `response_interval_ms` can probably be removed (check with @Lia-JX-Li ), as it can be derived from the timestamps.

start_of_turn is indeed more clear, so I changed it back.  I removed the response_interval_ms after consulting with Lia.

**Pull request merged**